### ISSUE TITLE
Include AzureML Datastore URIs in Docs

### ISSUE THREAD

**Issue by samuel100** [pytorch/data#945]
### 📚 The doc issue

It is possible to include in the [Working with cloud storage providers](https://pytorch.org/data/main/tutorial.html#working-with-cloud-storage-providers) section, some examples of AzureML datastore URIs? The AzureML datastore is a known implementation of fsspec (by installing the `azureml-fsspec` Python library). 

For example, loading from a tar file in an AzureML datastore:

```python
from torchdata.datapipes.iter import IterableWrapper

cifar_uri = "azureml://subscriptions/XXX/resourcegroups/XXX/workspaces/XXX/datastores/XXX/paths/cifar-10-python.tar.gz"

dp = IterableWrapper([cifar_uri]) \
        .open_files_by_fsspec(mode="rb") \
        .load_from_tar()

for path, filestream in dp:
    print(path)
```

AzureML Datastores handle the authentication to the storage layer (blob, ADLS) without the need for keys, secrets in code or environment variables.

### Suggest a potential alternative/fix

I would be happy to write the documentation.

**Comment by samuel100** [comment]
Created a PR #946  with documentation updates.

**Comment by ejguan** [comment]
cc: @NivekT

**Comment by NivekT** [comment]
Fixed by #946
